internal/graph/check.go

```diff
@@ -134,9 +129,9 @@ func (cc *ConcurrentChecker) checkInternal(ctx context.Context, req ValidatedChe
 	//
 	// If the filtering results in no further resource IDs to check, or a result is found and a single
 	// result is allowed, we terminate early.
-	foundResourceIds, filteredResourcesIds := filterForFoundMemberResource(req.ResourceRelation, req.ResourceIds, req.Subject)
-	if len(foundResourceIds) > 0 && req.DispatchCheckRequest.ResultsSetting == v1.DispatchCheckRequest_ALLOW_SINGLE_RESULT {
-		return checkResultsForResourceIds(foundResourceIds, emptyMetadata)
+	membershipSet, filteredResourcesIds := filterForFoundMemberResource(req.ResourceRelation, req.ResourceIds, req.Subject)
+	if membershipSet.HasDeterminedMember() && req.DispatchCheckRequest.ResultsSetting == v1.DispatchCheckRequest_ALLOW_SINGLE_RESULT {
```

Comment:
do we need to handle caveated members here?

Reply:
They can't be: the incoming resource cannot have a caveat, and the subject cannot have a caveat: this is a *direct* comparison if we reach a point where the check is occurring for the same resource and subject, ala `check group:something#member group:something#member`